diff --git a/es/clone-node.js b/es/clone-node.js
index 114cbf214f65cda237dbba67458c2b68fadf421e..2f31795bdb0f81b6a435b630a475b1ffe9f1d666 100644
--- a/es/clone-node.js
+++ b/es/clone-node.js
@@ -59,10 +59,6 @@ async function cloneChildren(nativeNode, clonedNode, options) {
     if (isSlotElement(nativeNode) && nativeNode.assignedNodes) {
         children = toArray(nativeNode.assignedNodes());
     }
-    else if (isInstanceOfElement(nativeNode, HTMLIFrameElement) &&
-        ((_a = nativeNode.contentDocument) === null || _a === void 0 ? void 0 : _a.body)) {
-        children = toArray(nativeNode.contentDocument.body.childNodes);
-    }
     else {
         children = toArray(((_b = nativeNode.shadowRoot) !== null && _b !== void 0 ? _b : nativeNode).childNodes);
     }
diff --git a/es/embed-webfonts.js b/es/embed-webfonts.js
index f6b0d96829c8a3301c142cacd07079e85b1f64fd..f5bfa86bc18c8314ad565c73c7dfab87728f3428 100644
--- a/es/embed-webfonts.js
+++ b/es/embed-webfonts.js
@@ -190,9 +190,12 @@ export async function embedWebFonts(clonedNode, options) {
         : options.skipFonts
             ? null
             : await getWebFontCSS(clonedNode, options);
-    if (cssText) {
+    const finalCssText = options.extraStyleContent != null
+        ? options.extraStyleContent.concat(cssText || '')
+        : cssText
+    if (finalCssText) {
         const styleNode = document.createElement('style');
-        const styleContent = document.createTextNode(cssText);
+        const styleContent = document.createTextNode(finalCssText);
         styleNode.appendChild(styleContent);
         if (clonedNode.firstChild) {
             clonedNode.insertBefore(styleNode, clonedNode.firstChild);
diff --git a/lib/clone-node.js b/lib/clone-node.js
index 214f4d30a50baae7f007d2399eeac298366538c7..4de9de97b39a620ebe8ba16e5feb9e552eb95bc6 100644
--- a/lib/clone-node.js
+++ b/lib/clone-node.js
@@ -134,10 +134,6 @@ function cloneChildren(nativeNode, clonedNode, options) {
                     if (isSlotElement(nativeNode) && nativeNode.assignedNodes) {
                         children = (0, util_1.toArray)(nativeNode.assignedNodes());
                     }
-                    else if ((0, util_1.isInstanceOfElement)(nativeNode, HTMLIFrameElement) &&
-                        ((_a = nativeNode.contentDocument) === null || _a === void 0 ? void 0 : _a.body)) {
-                        children = (0, util_1.toArray)(nativeNode.contentDocument.body.childNodes);
-                    }
                     else {
                         children = (0, util_1.toArray)(((_b = nativeNode.shadowRoot) !== null && _b !== void 0 ? _b : nativeNode).childNodes);
                     }
diff --git a/lib/embed-webfonts.js b/lib/embed-webfonts.js
index c91538bd9c3afa63326a91267fb7b5b9c999cccd..f3e3bc8f0e9e254b3c10e3c317d804e3c83f769e 100644
--- a/lib/embed-webfonts.js
+++ b/lib/embed-webfonts.js
@@ -300,9 +300,12 @@ function embedWebFonts(clonedNode, options) {
                     _c.label = 5;
                 case 5:
                     cssText = _a;
-                    if (cssText) {
+                    const finalCssText = options.extraStyleContent != null
+                        ? options.extraStyleContent.concat(cssText || '')
+                        : cssText
+                    if (finalCssText) {
                         styleNode = document.createElement('style');
-                        styleContent = document.createTextNode(cssText);
+                        styleContent = document.createTextNode(finalCssText);
                         styleNode.appendChild(styleContent);
                         if (clonedNode.firstChild) {
                             clonedNode.insertBefore(styleNode, clonedNode.firstChild);
diff --git a/src/clone-node.ts b/src/clone-node.ts
index 5dfcd117a1dd3d3d5cfb7c98ead3eb828085337a..8aa85ca88f84e8e64297673dcff36b21d48a95b4 100644
--- a/src/clone-node.ts
+++ b/src/clone-node.ts
@@ -88,11 +88,6 @@ async function cloneChildren<T extends HTMLElement>(
 
   if (isSlotElement(nativeNode) && nativeNode.assignedNodes) {
     children = toArray<T>(nativeNode.assignedNodes())
-  } else if (
-    isInstanceOfElement(nativeNode, HTMLIFrameElement) &&
-    nativeNode.contentDocument?.body
-  ) {
-    children = toArray<T>(nativeNode.contentDocument.body.childNodes)
   } else {
     children = toArray<T>((nativeNode.shadowRoot ?? nativeNode).childNodes)
   }
diff --git a/src/embed-webfonts.ts b/src/embed-webfonts.ts
index a84a699d37aeaaea19b9b31e0e98710f1e30da3c..6835efb8a2431aa8f066cd5ed0b47234b55f347a 100644
--- a/src/embed-webfonts.ts
+++ b/src/embed-webfonts.ts
@@ -258,9 +258,13 @@ export async function embedWebFonts<T extends HTMLElement>(
       ? null
       : await getWebFontCSS(clonedNode, options)
 
-  if (cssText) {
+  const finalCssText = options.extraStyleContent != null
+    ? options.extraStyleContent.concat(cssText || '')
+    : cssText
+
+  if (finalCssText) {
     const styleNode = document.createElement('style')
-    const styleContent = document.createTextNode(cssText)
+    const styleContent = document.createTextNode(finalCssText)
 
     styleNode.appendChild(styleContent)
 
diff --git a/src/types.ts b/src/types.ts
index 6023c3c2def1cb3ce83112c4c902997617582f59..b3cb4bd51f71ea708aab8bf616d8eef006b8dbf9 100644
--- a/src/types.ts
+++ b/src/types.ts
@@ -82,6 +82,11 @@ export interface Options {
    * create embed CSS for use across multiple calls to library functions.
    */
   fontEmbedCSS?: string
+  /**
+   * A CSS string to specify additional custom style content. It will be
+   * added to the style-tag.
+   */
+  extraStyleContent?: string
   /**
    * A boolean to turn off auto scaling for truly massive images..
    */
